test/topology/test_topology_failure_recovery.py

```diff
@@ -4,73 +4,105 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 #
 from test.pylib.manager_client import ManagerClient
-from test.pylib.internal_types import ServerInfo
 from test.pylib.scylla_cluster import ReplaceConfig
 from test.topology.conftest import skip_mode
+from test.topology.util import get_topology_coordinator
 import pytest
 import logging
-import asyncio
+import random
+import sys
 
 logger = logging.getLogger(__name__)
 
-async def inject_error_on(manager, error_name, servers):
-    errs = [manager.api.enable_injection(s.ip_addr, error_name, True) for s in servers]
-    await asyncio.gather(*errs)
+
+def init_random_seed():
+    """ Initialize random seed for the test.
+
+        The seed is logged so that the test can be reproduced if needed.
+    """
+    seed = random.randrange(sys.maxsize)
+    random.seed(seed)
+    logger.info("Random seed: %s", seed)
+
+
+async def get_running_servers(manager: ManagerClient):
+    """ Return the running servers in randomized order.
+
+        This helps to avoid making any assumptions on the order of the servers,
+        as in general the order of the cluster running servers can be somewhat
+        random (especially when initialized in parallel).
+    """
+    servers = await manager.running_servers()
+    random.shuffle(servers)
```

Comment:
The randomization has been done to make sure the problem is reproducible with reasonable rate (e.g. when run with "--repeat 100"). The actual fix is the `choose_non_coordinator` so the randomization could be removed eventually, but I think that it would actually make sense to keep it so that there can't be any assumptions made about the retrieved order of servers.

I'm also thinking whether it would make sense to even randomize the `manager.running_servers()` response - it would have the potential to break many tests, but in particular that would be the ones that might have a similar problem like this one (making assumptions about the server ordering).